spec/bin_packing/bin_spec.rb

```diff
@@ -35,4 +35,75 @@
       expect(bin.insert(BinPacking::Box.new(50, 100))).to eq true
     end
   end
+
+  describe 'insert_in_known_position' do
+    let(:bin) { BinPacking::Bin.new(100, 50) }
+
+    it 'should allow inserting a box without updating its position' do
+      box = BinPacking::Box.new(50, 50)
+      box.x = 40
+      box.y = 10
+
+      expect(bin.insert_in_known_position(box)).to eq(true)
+
+      expect(box.x).to eq(40)
+      expect(box.y).to eq(10)
+    end
+
+    it 'should not be movable by inserting new boxes' do
+      box = BinPacking::Box.new(50, 50)
+      box.x = 25
+      box.y = 25
+
+      bin.insert_in_known_position(box)
+
+      other_box = BinPacking::Box.new(30, 30)
+
+      expect(bin.insert(other_box)).to eq(false)
+      expect(other_box.packed?).to eq(false)
+    end
+
+    it 'should allow insertion of boxes if known boxes leave space' do
+      big_box = BinPacking::Box.new(50, 50)
+      big_box.x = 0
+      big_box.y = 0
+
+      bin.insert_in_known_position(big_box)
+
+      smaller_box = BinPacking::Box.new(30, 20)
+      smaller_box.x = 50
+      smaller_box.y = 0
+
+      bin.insert_in_known_position(smaller_box)
+
+      fitting_box = BinPacking::Box.new(20, 20)
+
+      expect(bin.insert(fitting_box)).to eq(true)
+
+      too_big_box = BinPacking::Box.new(40, 40)
+
+      expect(bin.insert(too_big_box)).to eq(false)
+    end
+
+    it 'should allow insertion of overlapping boxes' do
+      box_a = BinPacking::Box.new(50, 50)
+      box_a.x = 0
+      box_a.y = 0
+
+      box_b = BinPacking::Box.new(50, 50)
+      box_b.x = 5
+      box_b.y = 5
+
+      expect(bin.insert_in_known_position(box_a)).to eq(true)
+      expect(bin.insert_in_known_position(box_b)).to eq(true)
+    end
+
+    it 'should allow insertion of boxes too big for bins' do
```

Comment:
Interesting requirements 🙂

Reply:
The goal is to ensure no validations on this method. No matter how "wrong" the known position is, it should be used.